R/get-param.R

```diff
@@ -0,0 +1,128 @@
+#' Return theta values as a named vector
+#'
+#' @param .summary  A `bbi_nonmem_summary` or `bbi_summary_list` object.
+#' @export
+get_theta <- function(.summary){
+
+  if (inherits(.summary, "bbi_nonmem_summary")) {
+    .summary <- list(.summary)
+    check_model_object_list(.summary, .mod_types = c(NM_SUM_CLASS))
+  }else if(inherits(.summary, "bbi_summary_list")){
+    check_model_object(.summary, .mod_types = c("bbi_summary_list"))
+  }
+
+  thetas <- map(.summary, ~ {
+    # unpack bbi_summary_list element
+    if (!is.null(.x$bbi_summary)) .x <- .x$bbi_summary
+    param_names <- .x[[SUMMARY_PARAM_NAMES]]
+    theta <- .x[[SUMMARY_PARAM_DATA]][[length(.x[[SUMMARY_PARAM_DATA]])]]$estimates$theta
+    names(theta) <- param_names$theta
+    theta
+  })
+
+
+  if(length(thetas) == 1) thetas <- thetas[[1]]
+
+  return(thetas)
+}
+
+
+#' Return omega values as a matrix
+#'
+#' @param .summary  A `bbi_nonmem_summary` or `bbi_summary_list` object.
+#' @export
+get_omega <- function(.summary){
+
+  if (inherits(.summary, "bbi_nonmem_summary")) {
+    .summary <- list(.summary)
+    check_model_object_list(.summary, .mod_types = c(NM_SUM_CLASS))
+  }else if(inherits(.summary, "bbi_summary_list")){
+    check_model_object(.summary, .mod_types = c("bbi_summary_list"))
+  }
+
+  omegas <- map(.summary, ~ {
+    # unpack bbi_summary_list element
+    if (!is.null(.x$bbi_summary)) .x <- .x$bbi_summary
+
+    param_names <- .x[[SUMMARY_PARAM_NAMES]]
+    omega <- with(
```

Comment:
The `with` documentation recommends against using in the programming context:

```
Note:

     For _interactive_ use this is very effective and nice to read.
     For _programming_ however, i.e., in one's functions, more care is
     needed, and typically one should refrain from using ‘with()’, as,
     e.g., variables in ‘data’ may accidentally override local
     variables, see the reference.
```

(Same comment for `get_sigma`.)